spec/spec_helper.rb

```diff
@@ -0,0 +1,184 @@
+  #   # you configure your source control system to ignore this file.
+  #   config.example_status_persistence_file_path = "spec/examples.txt"
+  #
+  #   # Limits the available syntax to the non-monkey patched syntax that is
+  #   # recommended. For more details, see:
+  #   # https://relishapp.com/rspec/rspec-core/docs/configuration/zero-monkey-patching-mode
+  #   config.disable_monkey_patching!
+  #
+  #   # This setting enables warnings. It's recommended, but in some cases may
+  #   # be too noisy due to issues in dependencies.
+  #   config.warnings = true
+  #
+  #   # Many RSpec users commonly either run the entire suite or an individual
+  #   # file, and it's useful to allow more verbose output when running an
+  #   # individual spec file.
+  #   if config.files_to_run.one?
+  #     # Use the documentation formatter for detailed output,
+  #     # unless a formatter has already been configured
+  #     # (e.g. via a command-line flag).
+  #     config.default_formatter = "doc"
+  #   end
+  #
+  #   # Print the 10 slowest examples and example groups at the
+  #   # end of the spec run, to help surface which specs are running
+  #   # particularly slow.
+  #   config.profile_examples = 10
+  #
+  #   # Run specs in random order to surface order dependencies. If you find an
+  #   # order dependency and want to debug it, you can fix the order by providing
+  #   # the seed, which is printed after each run.
+  #   #     --seed 1234
+  #   config.order = :random
+  #
+  #   # Seed global randomization in this process using the `--seed` CLI option.
+  #   # Setting this allows you to use `--seed` to deterministically reproduce
+  #   # test failures related to randomization by passing the same `--seed` value
+  #   # as the one that triggered the failure.
+  #   Kernel.srand config.seed
+end
+
+HOME = File.expand_path(File.join(__dir__, "..")).freeze
+TEST_TMP = File.join(HOME, "monthly_inventory_test_temp").freeze
+
+CATALOG_PREP = File.join(TEST_TMP, "catalog_prep").freeze
+RIGHTS_DIR = File.join(TEST_TMP, "rights").freeze
+RIGHTS_DIR_ARCHIVE = File.join(TEST_TMP, "rights", "archive").freeze
+INGEST_BIBRECORDS = File.join(TEST_TMP, "ingest_bibrecords").freeze
+INGEST_BIBRECORDS_ARCHIVE = File.join(TEST_TMP, "ingest_bibrecords", "archive").freeze
+
+ENV["CATALOG_PREP"] = CATALOG_PREP
+ENV["RIGHTS_DIR"] = RIGHTS_DIR
+ENV["INGEST_BIBRECORDS"] = INGEST_BIBRECORDS
+
+MONTHLY_INVENTORY_LOCATIONS = [
+  CATALOG_PREP,
+  RIGHTS_DIR,
+  RIGHTS_DIR_ARCHIVE,
+  INGEST_BIBRECORDS,
+  INGEST_BIBRECORDS_ARCHIVE
+].freeze
+
+# Create monthly_inventory_test_temp directory
+# Create temp CATALOG_PREP, RIGHTS_DIR, INGEST_BIBRECORDS if necessary
+# Doing these tests in a separate directory that can be thrown away to
+# avoid stepping on the data/ directory before we can get this new Ruby
+# code better integrated.
+def init_test_tmp
+  Dir.mkdir(TEST_TMP) unless File.directory? TEST_TMP
```

Comment:
Rather than a static `TEST_TMP` used for all tests, I would rather use a structure like

```ruby
around(:each) do |example|
  Dir.mktmpdir do |tmpdir|
    setup_test_dirs(tmpdir)
    example.run
  end
end
```

This better isolates tests & automatically cleans up the temp directory after each test. (Running under Docker mitigates this somewhat assuming that the tmp directory is somewhere that isn't persistent between container runs).